### PR TITLE
Cherry-pick ORCA InPlaceUpdate implementation

### PR DESCRIPTION
**Pull request by foreyes**
Cherry-pick some ORCA patches from GreenPlum. Only ORCA SplitUpdate and the submissions it depends on are selected.

---

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by jiaqizho** [comment]
@foreyes  address the conflicts and make CI pass pls.

without conflicts address, the CI won't running.

**Comment by my-ship-it** [comment]
> @foreyes address the conflicts and make CI pass pls.
> 
> without conflicts address, the CI won't running.

@jiaqizho CI already passed, please continue to review.

File: src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 extern "C" {
```
Comment (jiaqizho) [inline src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:17]: no commit message in this commit.

File: src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp [hunk 2]
```diff
@@ -602,6 +602,10 @@ CTranslatorDXLToPlStmt::TranslateDXLTblScan(
 		plan->qual = qual;
 	}
 
+	if (md_rel->IsAORowOrColTable())
```
Comment (jiaqizho) [inline src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:606]: do we have checked the `ao_column`?

**Comment by jiaqizho** [comment]
@foreyes any updated?

File: src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp (hunk 2 above)
Comment (foreyes) [reply src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:606]: ao_column checked in IsAORowOrColTable()

File: src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp (hunk 1 above)
Comment (foreyes) [reply src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:17]: I'll add it later

**Comment by foreyes** [comment]
@jiaqizho @leborchuk 
This MR was not merged because it caused some testing failures in our internal repository, and I am still trying to find out the reason behind it.
But now there is no problem merging into the GitHub repository, and this MR only cherry picked some of the GreenPlum modifications while resolving code conflicts. Maybe I can figure out the internal repo CI failure later.

**Comment by foreyes** [comment]
I'll resolve the conflict and update this MR soon.

**Comment by foreyes** [comment]
@jiaqizho commit message updated and CI passed.

File: src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp [hunk 3]
```diff
@@ -602,6 +602,10 @@ CTranslatorDXLToPlStmt::TranslateDXLTblScan(
...
+	{
+		CheckSafeTargetListForAOTables(plan->targetlist);
```
Comment (jiaqizho) [inline src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:608]: I guest we need use the `RelationIsNonblockRelation` in this case?

The `PAX` am also not allow `System column` in target list.

**Review by jiaqizho — APPROVED** [review]
LGTM

**Pull request merged**